src/main/java/nl/knaw/dans/pf/language/ddm/api/Ddm2EmdCrosswalk.java

```diff
@@ -87,9 +87,27 @@ public EasyMetadata createFromValidated(final File file) throws CrosswalkExcepti
      * @throws CrosswalkException
      */
     public EasyMetadata createFrom(final String xml) throws CrosswalkException {
-        return validateEMD(walk(ddmValidator, xml, newTarget()));
+        return validateEMD(walk(ddmValidator, replaceSurrogatePairsWithTempCode(xml), newTarget()));
     }
 
+    private String replaceSurrogatePairsWithTempCode(String xmlStr) {
+        char[] chars = xmlStr.toCharArray();
+        StringBuilder sb = new StringBuilder(chars.length);
+
+        for (int i = 0; i < chars.length; ++i) {
+            if (Character.isHighSurrogate(chars[i])) {
+                if (i + 1 == chars.length)
+                    throw new IllegalStateException("String contains high surrogate pair at the end");
+                if (!Character.isSurrogatePair(chars[i], chars[i + 1]))
+                    throw new IllegalStateException("Invalid surrogate pair at position " + i);
+                sb.append("@@@SURROGATE-PAIR:").append(Character.codePointAt(chars, i)).append("@@@");
```

Comment:
please extract these strings into values, such that they can be reused in the rest of the code.